src/utils/ISoilUtils.C

```diff
@@ -0,0 +1,243 @@
+  {
+    backbone_strain[i].resize(number_of_points);
+    backbone_stress[i].resize(number_of_points);
+    modulus[i].resize(number_of_points);
+    for (std::size_t j = 0; j < number_of_points; ++j)
+    {
+      backbone_strain[i][j] = pow(10.0, -6.0 + 5.0 / (number_of_points - 1) * j);
+      Real theta =
+          theta_1[i] +
+          theta_2[i] * theta_4[i] *
+              pow(backbone_strain[i][j] / (taumax[i] / initial_shear_modulus[i]), theta_5[i]) /
+              (pow(theta_3[i], theta_5[i]) +
+               pow(backbone_strain[i][j] / (taumax[i] / initial_shear_modulus[i]), theta_5[i]));
+      modulus[i][j] =
+          (theta == 0)
+              ? 1.0 / (1.0 + backbone_strain[i][j] / (taumax[i] / initial_shear_modulus[i]))
+              : (1.0 / (backbone_strain[i][j] / (taumax[i] / initial_shear_modulus[i]))) *
+                    (0.5 / theta *
+                     (1.0 + backbone_strain[i][j] / (taumax[i] / initial_shear_modulus[i]) -
+                      sqrt(pow(1.0 + backbone_strain[i][j] / (taumax[i] / initial_shear_modulus[i]),
+                               2.0) -
+                           4.0 * theta * backbone_strain[i][j] /
+                               (taumax[i] / initial_shear_modulus[i]))));
+      backbone_stress[i][j] = modulus[i][j] * initial_shear_modulus[i] * backbone_strain[i][j];
+    }
+  }
+}
+
+void
+ISoilUtils::computeCoulombBackbone(std::vector<std::vector<Real>> & backbone_stress,
+                                   std::vector<std::vector<Real>> & backbone_strain,
+                                   const std::vector<unsigned int> & layer_ids,
+                                   const std::vector<Real> & initial_shear_modulus,
+                                   const std::vector<Real> & friction_coefficient,
+                                   const std::vector<Real> & hardening_ratio,
+                                   const std::vector<Real> & p_ref,
+                                   const std::string & name)
+{
+  backbone_strain.resize(layer_ids.size());
+  backbone_stress.resize(layer_ids.size());
+  for (std::size_t i = 0; i < layer_ids.size(); i++)
+  {
+    backbone_strain[i].resize(2);
+    backbone_stress[i].resize(2);
+    if (hardening_ratio[i] >= 1.0)
+      mooseError("Error in " + name + ". hardening_ratio cannot be greater than or equal to 1.0.");
+    backbone_strain[i][0] = p_ref[i] * friction_coefficient[i] / initial_shear_modulus[i];
+    backbone_stress[i][0] = p_ref[i] * friction_coefficient[i];
+    backbone_strain[i][1] = 100.0;
+    backbone_stress[i][1] = backbone_stress[i][0] +
+                            (backbone_strain[i][1] - backbone_strain[i][0]) * hardening_ratio[i] *
+                                initial_shear_modulus[i];
+  }
+}
+
+void
+ISoilUtils::computeSoilLayerProperties(
+    std::vector<std::vector<Real>> & youngs,
+    std::vector<std::vector<Real>> & yield_stress, // *** YIELD STRAIN, NOT YIELD STRESS***
```

Comment:
@sveerara Do you know the purpose of this comment?

Reply:
@aeslaughter and @cbolisetti, In ComputeIsoilStress.C we are diving the backbone stress-strain curve into multiple elastic-perfectly plastic curves each characterized by a young's modulus and a yield stress.  The young's modulus is used in the calculation of the elasticity tensor (which is a 4th order tensor with 81 components stored at each qp) that varies linearly with change in Young's modulus. So instead of defining N (number of elastic-perfectly plastic curves) elasticity tensors, I am defining only one elasticity tensor which is calculated with a young's modulus of 1. And all the other parameters (like yield stress, deviatoric stress, effective stress, etc) are divided by the Young's modulus so that the calculation can proceed with just this one elasticity tensor. 

Since yield stress/young's modulus = yield strain, it is technically the yield strain that is used in the calculation. But since it is used in the yield condition calculation (line 311-312 of ComputeISoilStress.C), I just maintained the name `yield stress` instead of `yield strain`. All the terms in the yield condition calculation are basically normalized by the Young's modulus.

I am okay with changing the name to yield strain if that is less confusing.

Reply:
@moosebuild Correction do you know why you called this stress, @cbolisetti is indicates that this should be stress. I want to get the name correct before this is merged.

Reply:
@aeslaughter, I called it stress because the yield condition is usually defined in terms of yield stress and not yield strain and I used this parameter in the yield condition calculation. 

Maybe I can rework the ComputIsoilStress.C to clean up the names. Looks like normalizing by Young's modulus is causing a lot of naming confusion.

Reply:
@sveerara I now see why you call it yield_stress. However, I do think this is a bit confusing, and will be more so when we come back to it a few months/years later. If you could rework this, that would be great.

Reply:
@aeslaughter and @cbolisetti , if its okay with both of you, I will pull in Chandu's changes and then rework ComputeISoilStress such that _yield_stress corresponds to yield stress instead of strain and deviatoric stress and effective stress correspond to deviatoric and effective stress instead of stress normalized by Young's modulus.

Reply:
Sounds good. I will do that.